### ISSUE TITLE
Terminal freezes after setting @ray.remote(num_gpu=2)

### ISSUE THREAD

**Issue by ProNoobLi** [ray-project/ray#6418]
### What is the problem?
Terminal freezes after setting the gpu numbers by @ray.remote(num_gpu=2). However, everything works with @ray.remote as a decorator of my class  except training speed.

*Ray version and other system information (Python version, TensorFlow version, OS):*
Ray: 0.8.0dev 
TF: 2.0.0
Ubuntu: 16.04
Python 3.6

*Does the problem occur on the [latest wheels](https://ray.readthedocs.io/en/latest/installation.html)?*
Yes

### Reproduction
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):
![image](https://user-images.githubusercontent.com/33815430/70526771-4f1b2200-1b85-11ea-87c2-10a186259bb1.png)
![image](https://user-images.githubusercontent.com/33815430/70526930-a28d7000-1b85-11ea-8e9a-3e108f63cbeb.png)
When I terminate it, error reports that the code gets stucked at ray._raylet.check_status

```python
from __future__ import absolute_import, division, print_function, unicode_literals
from tensorflow.keras import layers
from tensorflow.keras.layers import Dense, Flatten, Conv2D
from tensorflow.keras import Model

import time
import ray
import numpy as np

EPOCHS = 5

ray.shutdown()
ray.init(num_cpus=20, include_webui=False, ignore_reinit_error=True)


class MyModel(Model):
  def __init__(self):
    super(MyModel, self).__init__()
    self.conv1 = Conv2D(32, 3, activation='relu')
    self.flatten = Flatten()
    self.d1 = Dense(128, activation='relu')
    self.d2 = Dense(10, activation='softmax')

  def call(self, x):
    x = self.conv1(x)
    x = self.flatten(x)
    x = self.d1(x)
    return self.d2(x)


@ray.remote(num_gpus=4)
class Train():
  def __init__(self):
    import tensorflow as tf
    mnist = tf.keras.datasets.mnist
    (x_train, y_train), (x_test, y_test) = mnist.load_data()
    x_train, x_test = x_train / 255.0, x_test / 255.0

    # Add a channels dimension
    x_train = x_train[..., tf.newaxis]
    x_test = x_test[..., tf.newaxis]


    self.train_ds = tf.data.Dataset.from_tensor_slices(
        (x_train, y_train)).shuffle(10000).batch(32)

    self.test_ds = tf.data.Dataset.from_tensor_slices((x_test, y_test)).batch(32)

    # Create an instance of the model
    self.model = MyModel()
    self.loss_object = tf.keras.losses.SparseCategoricalCrossentropy()
    self.optimizer = tf.keras.optimizers.Adam()


    self.train_loss = tf.keras.metrics.Mean(name='train_loss')
    self.train_accuracy = tf.keras.metrics.SparseCategoricalAccuracy(name='train_accuracy')
    self.test_loss = tf.keras.metrics.Mean(name='test_loss')
    self.test_accuracy = tf.keras.metrics.SparseCategoricalAccuracy(name='test_accuracy')

    @tf.function
    def train_step(images, labels):
      import tensorflow as tf
      with tf.GradientTape() as tape:
        predictions = self.model(images)
        loss = self.loss_object(labels, predictions)
      gradients = tape.gradient(loss, self.model.trainable_variables)
      self.optimizer.apply_gradients(zip(gradients, self.model.trainable_variables))

      self.train_loss(loss)
      self.train_accuracy(labels, predictions)

    @tf.function
    def test_step(images, labels):
      import tensorflow as tf
      predictions = self.model(images)
      t_loss = self.loss_object(labels, predictions)

      self.test_loss(t_loss)
      self.test_accuracy(labels, predictions)

    self.tf_train_step = train_step
    self.tf_test_step = test_step


  def train(self):
    import tensorflow as tf
    for epoch in range(EPOCHS):
      for images, labels in self.train_ds:
        self.tf_train_step(images, labels)

      for test_images, test_labels in self.test_ds:
        self.tf_test_step(test_images, test_labels)

      template = 'Epoch {}, Loss: {}, Accuracy: {}, Test Loss: {}, Test Accuracy: {}'
      print(template.format(epoch+1,
                            self.train_loss.result(),
                            self.train_accuracy.result()*100,
                            self.test_loss.result(),
                            self.test_accuracy.result()*100))

      # Reset the metrics for the next epoch
      self.train_loss.reset_states()
      self.train_accuracy.reset_states()
      self.test_loss.reset_states()
      self.test_accuracy.reset_states()

  def get_weights(self):
    return self.model.get_weights()

  def set_weights(self, weights):
    return self.model.set_weights(weights)


if __name__ == "__main__":
    time.sleep(1)
    start = time.time()

    my_train_actor1 = Train.remote()
    my_train_actor2 = Train.remote()   
    
    weights = ray.get([my_train_actor1.get_weights.remote(),
                       my_train_actor2.get_weights.remote() 
                       ])

    ave_weights = [(layer1 + layer2) / 2 for layer1, layer2 in zip(weights[0], weights[1])] 
    
    weight_id = ray.put(ave_weights)

    [
    actor.set_weights.remote(weight_id) 
    for actor in [my_train_actor1, my_train_actor2]
    ]
    

    ray.get([actor.train.remote() for actor in [my_train_actor1, my_train_actor2]])

    end = time.time()
    print('This took {} seconds'.format(end - start))
```
If we cannot run your script, we cannot fix your issue.

**Comment by edoakes** [comment]
Looks like this is hanging in `ray.get()` somewhere. Is it possible that there aren't enough resources to run a task or a task is never finishing?

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
